src/eth/index.ts

```diff
@@ -80,6 +101,41 @@ export class ETH extends EventEmitter {
     this.emit('message', code, payload)
   }
 
+  /**
+   * Eth 64 Fork ID validation (EIP-2124)
+   * @param forkId Remote fork ID
+   */
+  _validateForkId(forkId: Buffer[]) {
+    const c = this._peer._common
+
+    const peerForkHash = `0x${forkId[0].toString('hex')}`
+    const peerNextFork = buffer2int(forkId[1])
+
+    if (this._forkHash === peerForkHash) {
+      if (peerNextFork) {
```

Comment:
Style nit: assumes knowledge that `0` is regarded as `false`.

Reply:
Updated (next push).

Reply:
Are you otherwise confident that the validation rules (see equally named section in the [EIP-2124 specs](https://eips.ethereum.org/EIPS/eip-2124#specification) are adequately covered? I had my problems here since the rules are stating the success cases and are very vague on the error cases ("Reject in all other cases") and I had to do somewhat of the inverse (implement the error cases and succeed in all other cases 😋 ) way around.

A bit unfortunate that this is not more exactly covered by the spec, would assume it wouldn't be that hard to just state *all* possible cases there.

Reply:
I will re-check this 👍